src/foam/nanos/crunch/UserCapabilityJunctionDAO.js

```diff
@@ -99,16 +104,100 @@ foam.CLASS({
       javaCode: `
       checkOwnership(x, obj);
 
+      // if the junction is being uppdated from GRANTED to EXPIRED, put into junctionDAO without checking prereqs and data
+      UserCapabilityJunction old = (UserCapabilityJunction) getDelegate().find_(x, ((UserCapabilityJunction) obj).getId());
+      if ( old.getStatus() == CapabilityJunctionStatus.GRANTED && ((UserCapabilityJunction) obj).getStatus() == CapabilityJunctionStatus.EXPIRED ) 
+        return getDelegate().put_(x, obj);
+
       boolean prereq = checkPrereqs(x, obj);
       boolean data = validateData(x, obj);
 
-      if ( prereq && data ) ((UserCapabilityJunction) obj).setStatus(CapabilityJunctionStatus.GRANTED);
+      if ( prereq && data ) {
+        ((UserCapabilityJunction) obj).setStatus(CapabilityJunctionStatus.GRANTED);
+        saveDataToDAO(x, ((UserCapabilityJunction) obj));
+        configureJunctionExpiry(x, (UserCapabilityJunction) obj, old);
+      }
       else ((UserCapabilityJunction) obj).setStatus(CapabilityJunctionStatus.PENDING);
 
       return getDelegate().put_(x, obj);
       
       `
     },
+    {
+      name: 'saveDataToDAO',
+      args: [
+        {
+          name: 'x',
+          type: 'Context'
+        },
+        {
+          name: 'obj',
+          type: 'foam.nanos.crunch.UserCapabilityJunction'
+        }
+      ],
+      documentation: `save the UserCapabilityJunction data to its dao, if daoKey is provided in the Capability model`,
```

Comment:
Please provide more of an explanation here so that's it's clear why we need to do this. Maybe include an example as well. This is what Kevin, you and I talked about in person.